plugins/content/caption/base/lib/model/CaptionAsset.php

```diff
@@ -70,6 +73,39 @@ public function getName()
 		return $this->getLanguage();
 	}
 	
+	public function preUpdate(PropelPDO $con = null)
+	{
+		$ret = parent::preUpdate($con); // TODO: Change the autogenerated stub
+		
+		if(($this->isColumnModified(assetPeer::STATUS) && $this->getStatus() == CaptionAsset::ASSET_STATUS_READY && $this->getAccuracy() && $this->getLanguage()) ||
```

Comment:
can be improved - 
$this->getStatus() == CaptionAsset::ASSET_STATUS_READY && (A or B or C)